packages/graphcache-rn-async-storage/src/makeAsyncStorage.ts

```diff
@@ -0,0 +1,114 @@
+import { StorageAdapter } from '@urql/exchange-graphcache';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import NetInfo from '@react-native-community/netinfo';
+
+export type StorageOptions = {
+  dataKey?: string;
+  metadataKey?: string;
+  maxAge?: number; // Number of days
+};
+
+let disconnect;
+
+export const makeAsyncStorage: (ops?: StorageOptions) => StorageAdapter = ({
+  dataKey = 'graphcache-data',
+  metadataKey = 'graphcache-metadata',
+  maxAge = 7,
+} = {}) => {
+  const todayDayStamp = Math.floor(
+    new Date().valueOf() / (1000 * 60 * 60 * 24)
+  );
+  const todayBatch = {};
+  const prefixCheck = new RegExp(`^${dataKey}_(\\d+)$`);
+
+  return {
+    /**
+     * On initial read, pull storage keys and see which match our storage key pattern.
+     * Any record that is expired should be removed from AsyncStorage.
+     * The rest of the batches should be merged into a cache in chrono order.
+     * Also hydrate todayBatch if we find it.
+     */
+    readData: async () => {
+      const cache = {};
+
+      try {
+        const persistedDayStamps = (await AsyncStorage.getAllKeys())
+          .reduce<number[]>((filtered, curr) => {
+            const match = curr.match(prefixCheck);
+            const ts = match ? match[1] : undefined;
+
+            if (ts) {
+              filtered.push(Number(ts));
+            }
+            return filtered;
+          }, [])
+          .sort();
+
+        // eslint-disable-next-line es5/no-for-of
+        for (const dayStamp of persistedDayStamps) {
+          // Discard
+          if (todayDayStamp - dayStamp > maxAge) {
+            await AsyncStorage.removeItem(`${dataKey}_${dayStamp}`);
+          }
+          // Parse batch and merge in.
+          else {
```

Comment:
nit: hard to read, since the comment is superficial (like "Discard") and this one separates the `if` from `else`

Reply:
I've changed this logic completely so that else is now gone